### ISSUE TITLE
loss nan error when set filter_empty_gt=False

### ISSUE THREAD

**Issue by zhengye1995** [shinya7y/UniverseNet#14]
Hello, thanks for your outstanding work!

I want to train universeNet with the images which do not contain objects, so I set filter_empty_gt=False, but the loss is nan.

I have reduced the learning rate and tried to increase the warm-up step, but they are not work.

I have also tried to train my data with other models like cascade rcnn or faster rcnn with the same setting, the losses for them are both normal.

So, how can i fix this issue?

**Comment by shinya7y** [comment]
A possible solution is applying this fix https://github.com/open-mmlab/mmdetection/pull/3702
to [gfl_head.py](https://github.com/shinya7y/UniverseNet/blob/master/mmdet/models/dense_heads/gfl_head.py).

[if len(pos_inds) == 0: weight_targets become 0](https://github.com/shinya7y/UniverseNet/blob/0ad00250a40797e1ffe43c413a8058ffce74a814/mmdet/models/dense_heads/gfl_head.py#L301),
avg_factor may become 0, resulting in nan.

**Comment by whut2962575697** [comment]
Hello，do you fix the  issue? https://github.com/shinya7y/UniverseNet/issues/13

**Comment by shinya7y** [comment]
I updated the master branch with the abovementioned fix.
If the issue persists, please let me know more details.

**Comment by zhengye1995** [comment]
> I updated the master branch with the abovementioned fix.
> If the issue persists, please let me know more details.

Thanks for your replay, it works. But I still need to reduced the learning rate and tried to increase the warm-up step, which may results in the performance is not well.

**Comment by shinya7y** [comment]
I close this issue.
If there are performance issues (e.g., lower AP than the baselines ATSS and GFL), please open another issue.
